libafl/src/stages/mutational.rs

```diff
@@ -94,15 +92,15 @@ where
     fn mutator_mut(&mut self) -> &mut M;
 
     /// Gets the number of iterations this mutator should run for.
-    fn iterations(&self, state: &mut Self::State) -> Result<usize, Error>;
+    fn iterations(&self, state: &mut S) -> Result<usize, Error>;
 
     /// Runs this (mutational) stage for the given testcase
     #[allow(clippy::cast_possible_wrap)] // more than i32 stages on 32 bit system - highly unlikely...
     fn perform_mutational(
         &mut self,
         fuzzer: &mut Z,
         executor: &mut E,
-        state: &mut Self::State,
+        state: &mut S,
         manager: &mut EM,
     ) -> Result<(), Error> {
```

Comment:
This whole method should be removed from the trait definition and instead made as a generic function that mutational stages can call.

Then, we can `impl<M, ...> Stage<...> for M where M: MutationalStage<...>`.